### DISCUSSION TITLE
What technology does Meshcentral's default web based screen sharing uses?

### DISCUSSION THREAD

**Discussion by imtadashi** [Ylianst/MeshCentral#5117] [Q&A] [answered]
Hi, I would like to learn about the technology Meshcentral uses for its default web based screen sharing. I have been using Meshcentral for a while and I have found that WebSocket and WebRTC are used for bi-directional communication. VNC and RDP is another options available. But what technology does it use for default screen sharing. At first I thought it was WebRTC, but I think WebRTC does not have the ability to share screen when the system is locked or signed out. In Meshcentral's default web based screen sharing, even locked screen was able to be viewed. Can I please know what technology does it use for this function?

**Comment by si458** [comment]
Everything is websocket

But you can enable webrtc for better performance of the screen viewing

**Reply by jsastriawan** [reply, marked as answer]
It is based on JPEG tile stream. So basically the agent takes a screenshot, then slice it into multiple tiles of image (32x32 pixel), compress them into JPEG. Initially all tiles will be sent, the next cycle will be the changed tiles.

**Comment by imtadashi** [comment]
Thank you so much for your response. I will delve deeper into the topic to expand my understanding.
